app/controllers/api/accounts_controller.rb

```diff
@@ -0,0 +1,65 @@
+class Api::AccountsController < ApplicationController
+  def new
+    @user = current_user
+  end
+
+  def create
+    @user = current_user
+    if valid_form?
+      UserApi.save_key(api_request_params, @user.id)
+      flash[:success] = "API Key Request Accepted"
+      redirect_to api_accounts_dashboard_path
+    else
+      flash.now[:error] = set_error_message.join(", ")
+      render :new
+    end
+  end
+
+  def show
+    @user_api = UserApi.find_by(uid: current_user.id)
+  end
+
+  def overwrite
+    @user = current_user
+    if valid_overwrite_form?
+      UserApi.find_by(uid: @user.id).overwrite_key
+      flash[:success] = "Your old API key has been successfully overwritten"
+      redirect_to api_accounts_dashboard_path
+    else
+      flash.now[:error] = "You entered the incorrect password"
+      render :new
+    end
+  end
+
+
+  private
+    def valid_form?
+      !params[:first_name].nil? &&
+      !params[:last_name].nil? &&
+      !params[:email].nil? &&
+      !params[:url].nil? &&
+      !params[:redirect_url].nil? &&
```

Comment:
`params[:redirect_url].present?` should do the same things as `!params[:redirect_url].nil?` and is easier to read.